sky/jobs/core.py

```diff
@@ -53,7 +54,11 @@ def launch(
     entrypoint = task
     dag_uuid = str(uuid.uuid4().hex[:4])
 
-    dag = dag_utils.convert_entrypoint_to_dag_and_apply_policy(entrypoint)
+    dag = dag_utils.convert_entrypoint_to_dag(entrypoint)
+    # TODO(zhwu): We should only apply policy to dag and save the config file,
+    # instead of having the config file actually being used.
+    dag, mutated_user_config = policy_utils.apply(dag,
+                                                  apply_skypilot_config=False)
```

Comment:
I found this arg hard to understand in call sites.

"use_mutated_skypilot_config" / "use_mutated_config_in_current_request" (latter seems clearer to me)? 

the two "apply" don't mean the same thing, so "use" seems better.

Reply:
Good point! How about: `update_skypilot_config_for_current_request`?